llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -962,22 +962,35 @@ void VPWidenIntrinsicRecipe::execute(VPTransformState &State) {
 
   // Use vector version of the intrinsic.
   Module *M = State.Builder.GetInsertBlock()->getModule();
+  bool IsVPIntrinsic = VPIntrinsic::isVPIntrinsic(VectorIntrinsicID);
   Function *VectorF =
-      Intrinsic::getOrInsertDeclaration(M, VectorIntrinsicID, TysForDecl);
-  assert(VectorF && "Can't retrieve vector intrinsic.");
+      IsVPIntrinsic
+          ? VPIntrinsic::getOrInsertDeclarationForParams(M, VectorIntrinsicID,
+                                                         TysForDecl[0], Args)
+          : Intrinsic::getOrInsertDeclaration(M, VectorIntrinsicID, TysForDecl);
+  assert(VectorF &&
+         "Can't retrieve vector intrinsic or vector-predication intrinsics.");
 
-  auto *CI = cast_or_null<CallInst>(getUnderlyingValue());
   SmallVector<OperandBundleDef, 1> OpBundles;
-  if (CI)
-    CI->getOperandBundlesAsDefs(OpBundles);
+  if (!IsVPIntrinsic) {
+    if (auto *CI = cast_or_null<CallInst>(getUnderlyingValue()))
+      CI->getOperandBundlesAsDefs(OpBundles);
```

Comment:
Please add a comment on why this is isn't supported for VPINstrinsics.

Reply:
If I pass the Instuction  in vplantransform, it will definitely fail here.
![image](https://github.com/user-attachments/assets/add673b9-b4c8-47b9-a53c-820cd49d2624)